File: test-fixtures/codebase/sample.go
// Package sample provides example Go code for testing.
package sample

import "fmt"

// Greeter is an exported struct for greeting users.
// It holds a name to use in greetings.
type Greeter struct {
	name string
}

// unexportedStruct is private to this package.
type unexportedStruct struct {
	value int
}

// NewGreeter creates a new Greeter instance.
// This is a constructor function.
func NewGreeter(name string) *Greeter {
	return &Greeter{name: name}
}

// Greet returns a greeting message.
func (g *Greeter) Greet() string {
	return fmt.Sprintf("Hello, %s!", g.name)
}

// privateHelper is not exported.
func privateHelper() int {
	return 42
}

// Add adds two integers together.
func Add(a, b int) int {
	return a + b
}

// multiply is a private function.
func multiply(a, b int) int {
	return a * b
}

// ProcessData handles data processing.
// It takes a map and returns processed results.
func ProcessData(data map[string]interface{}) map[string]interface{} {
	result := make(map[string]interface{})
	for k, v := range data {
		result[k] = v
	}
	return result
}

File: test-fixtures/codebase/Sample.cs
using System;

namespace Sample
{
    /// <summary>
    /// Sample C# class for testing.
    /// Demonstrates various C# features.
    /// </summary>
    public class Greeter
    {
        private string _name;

        /// <summary>
        /// Creates a new Greeter instance.
        /// </summary>
        /// <param name="name">The name to greet.</param>
        public Greeter(string name)
        {
            _name = name;
        }

        /// <summary>
        /// Returns a greeting message.
        /// </summary>
        /// <returns>A greeting string.</returns>
        [Obsolete("Use GreetFormal instead")]
        public string Greet()
        {
            return $"Hello, {_name}!";
        }

        /// <summary>
        /// Returns a formal greeting.
        /// </summary>
        public string GreetFormal()
        {
            return $"Greetings, {_name}.";
        }

        private void InternalMethod()
        {
            // Private method
        }
    }

    /// <summary>
    /// Internal helper class.
    /// </summary>
    internal class PrivateHelper
    {
        public static int Add(int a, int b)
        {
            return a + b;
        }

        private static int Multiply(int a, int b)
        {
            return a * b;
        }
    }

    /// <summary>
    /// Public interface for services.
    /// </summary>
    [Serializable]
    public interface IService
    {
        /// <summary>
        /// Executes the service.
        /// </summary>
        void Execute();
    }

    /// <summary>
    /// Data processor class.
    /// </summary>
    public static class DataProcessor
    {
        /// <summary>
        /// Processes input data.
        /// </summary>
        [System.Diagnostics.Conditional("DEBUG")]
        public static void Process(string data)
        {
            Console.WriteLine(data);
        }
    }
}
